docs/adrs/ADR-1.md

```diff
@@ -13,29 +12,97 @@ Deepansh
 ## Abstract
 > Abstract is a multi-sentence (short paragraph) technical summary. This should be a very brief, to the point and human-readable version of the document section. Someone should be able to read only the abstract to get the gist of what this document is about in its current state.
 
+A cron job (running on AWS EC2) will go through the [**Quiz Session Creator**](https://docs.google.com/spreadsheets/d/1viSIQZ3oKvZGFu08YetZs9l4aZSMGAk4aeyo-hvp8SQ/edit#gid=0) google sheet every 15 minutes.
+On every run, the cron job will select the sessions with the following filters:
+- Sessions which are either infinite sessions or non infinite sessions which have ended
+- Sessions with `test_type` as `homework`
+
+The selected session ids will be sent to an SNS notification, which will trigger a Lambda function implementing the ETL flow. The ETL flow will extract the required data, transform it, and dump into the required tables.
+
 ## Context & Problem Description
 > Go into detail about the subject in question. Why is this decision important? What problems are there that this decision needs to be made? The urgency of the decision? Datapoints and related background information. Vocabulary and key terms.
 
+Till now we have only been syncing assessment sessions but not homework sessions. For homework sessions, the extracted data will come from two places: **Firestore Sessions DB** and **Quiz Mongo DB**.
+The data being extracted from the **Firestore Sessions DB** is the data pertaining to the sessions. Things like `program`, `grade`, `course`, `stream` and many other, basically the column values stored in the **Quiz Session Creator** google sheet. The data being extracted from the **Quiz Mongo DB** will be the responses of students who have attempted that quiz and the related metadata. Decisions regarding some questions mentioned below need to be taken:
+1. Where should the homework sessions data be stored on BQ? Should it be stored in the `test_responses` table that exists or some new table?
```

Comment:
It depends on the use case completely. Not on tech simplicity or anything else. If we predict that analysis for tests and homeworks will be mostly done together, then they should be in the same table. If not, then separate.

Here my intuition is that they should be separate (even if columns are the same). Analysis will be *mostly* for tests separately and homeworks separately (I say *mostly*, because for things like student profile, we'll get both), so for the analysis this separation will be done with filters. Instead of this, it seems better to me to just keep the separation here itself and do the concatenation for the few use cases where it's needed